src/Microsoft.SqlTools.ServiceLayer/Migration/MigrationService.cs

```diff
@@ -290,19 +288,98 @@ await requestContext.SendResult(new ValidateWindowsAccountResult()
                 {
                     int ret = Marshal.GetLastWin32Error();
                     string errorMessage = new Win32Exception(Marshal.GetLastWin32Error()).Message;
-                    await requestContext.SendResult(new ValidateWindowsAccountResult()
-                    {
-                        Success = returnValue,
-                        ErrorMessage = errorMessage
-                    });
-                    return;
+                    await requestContext.SendError(errorMessage);
                 }
+                else
+                {
+                    await requestContext.SendResult(true);
+                }
+            } 
+            else 
+            {
+                await requestContext.SendResult(true);
             }
-            await requestContext.SendResult(new ValidateWindowsAccountResult()
+        }
+
+        internal async Task HandleValidateNetworkFileShareRequest(
+            ValidateNetworkFileShareRequestParams parameters,
+            RequestContext<bool> requestContext)
+        {
+            if (!ValidateWindowsDomainUsername(parameters.Username))
             {
-                Success = true,
-                ErrorMessage = ""
-            });
+                await requestContext.SendError("Invalid user name format. Example: Domain\\username");
+                return;
+            }
+
+            if (!ValidateUNCPath(parameters.Username))
+            {
+                await requestContext.SendError("Invalid network share path. Example: \\\\Servername.domainname.com\\Backupfolder");
+                return;
+            }
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                int separator = parameters.Username.IndexOf("\\");
+                string domainName = (separator > -1) ? parameters.Username.Substring(0, separator) : string.Empty;
```

Comment:
You've already validated that the username is in this format so these checks are unnecessary.

And anyways - as above if you return the match results from the ValidateWindowsDomainUsername method then you can just use that instead of substringing it.

Reply:
Sorry these are for 2 different screens. 

1. We are using the validate windows username for this screen. (To get the password for the integrated login username).
<img width="828" alt="image" src="https://user-images.githubusercontent.com/6816294/117055657-3a867f00-acd0-11eb-861c-fa3924ad342c.png">

2. File share validation is done in this page. We verify whether the ADS client is able to connect to the remote file share using the information provided in this page. It can be the same account from 1 but there is a high chance that the user can setup a file share on remote machine). 
<img width="751" alt="image" src="https://user-images.githubusercontent.com/6816294/117055872-876a5580-acd0-11eb-9332-407240248a3c.png">